src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java

```diff
@@ -0,0 +1,75 @@
+package com.internet.shop.web.filter;
+
+import com.internet.shop.controllers.LoginController;
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorizationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/user/all",List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/all",List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add",List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/delete",List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/product/all",List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/buy",List.of(Role.RoleName.USER));
+        protectedUrls.put("/cart",List.of(Role.RoleName.USER));
+        protectedUrls.put("/cart/delete",List.of(Role.RoleName.USER));
+        protectedUrls.put("/order/complete",List.of(Role.RoleName.USER));
+        protectedUrls.put("/user/order/details",List.of(Role.RoleName.USER));
+        protectedUrls.put("/order/delete",List.of(Role.RoleName.USER));
+        protectedUrls.put("/user/orders",List.of(Role.RoleName.USER));
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+            FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String requestedUrl = req.getServletPath();
+        if (!protectedUrls.containsKey(requestedUrl)) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long userId = (Long) req.getSession().getAttribute(LoginController.USER_ID);
+        User currentUser = userService.get(userId);
+        if (isAuthorized(currentUser, protectedUrls.get(requestedUrl))) {
```

Comment:
let's put both conditions in one `if`

Reply:
Done

Reply:
I cant put both conditions in one if because the first condition provides "NoSuchElementException" error protection when you get a user on the first launch of the program. Having done this, I will need to work out one more error that will not be optimization. Left as it was.

Reply:
@AndrewPolishchuc ок, соглашусь, давай оставим как есть